src/plugins/query_enhancements/public/search/ppl_search_interceptor.ts

```diff
@@ -88,21 +95,43 @@ export class PPLSearchInterceptor extends SearchInterceptor {
     return this.runSearch(request, options.abortSignal, strategy);
   }
 
-  private buildQuery() {
-    const { queryString } = this.queryService;
-    const query: Query = queryString.getQuery();
-    const dataset = query.dataset;
-    if (!dataset || !dataset.timeFieldName) return query;
-    const datasetService = queryString.getDatasetService();
-    if (datasetService.getType(dataset.type)?.languageOverrides?.PPL?.hideDatePicker === false)
-      return query;
-    // Only append time range if query is running search command
+  private buildQuery(request: IOpenSearchDashboardsSearchRequest) {
+    // Use query from request if available, otherwise fall back to queryStringManager
+    const query =
+      request.params?.body?.query?.queries?.[0] || this.queryService.queryString.getQuery();
+    // Only append filters if query is running search command (e.g. not describe command)
     if (!isPPLSearchQuery(query)) return query;
 
-    const [baseQuery, ...afterPipeParts] = query.query.split('|');
-    const afterPipe = afterPipeParts.length > 0 ? ` | ${afterPipeParts.join('|').trim()}` : '';
-    const timeFilter = this.getTimeFilter(dataset.timeFieldName);
-    return { ...query, query: baseQuery + timeFilter + afterPipe };
+    const filters = this.queryService.filterManager.getFilters();
+    const index = request.params?.index
+      ? this.indexPatterns.getByTitle(request.params.index)
```

Comment:
do we need to ignore errors?

Reply:
however if we do want to move to a data source specific search interceptor and strategy. then this might influence where this should exist